portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,86 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+
+//adds a random fact to the page
+function addRandomFact(){
+    const randomFacts=
+        ['I live in Manhattan, New York', 'I have the world\'s sleepiest dog', 'I absolutely drank too much coffee today'];
+
+    //Pick a random fact
+    const randomFact=randomFacts[Math.floor(Math.random() * randomFacts.length)];
+
+    //Add it to the page
+    const factContainer = document.getElementById('fact-container');
+    factContainer.innerText=randomFact;
+}
+
+function getRandomQuote() {
+  console.log('Fetching a random quote.');
+
+  // The fetch() function returns a Promise because the request is asynchronous.
+  const responsePromise = fetch('/data');
+
+  // When the request is complete, pass the response into handleResponse().
+  responsePromise.then(handleResponse);
```

Comment:
can you handle error case? e.g. 
`.catch((error) => {
  console.error('Error:', error);
});`
[full api](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch)